compiler bug when a pointer indirection is used on an interface (which is invalid)

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: macos, macOS, 12.0.1, 21A559
Processor: 10 cpus, 64bit, little endian, Apple M1 Max
CC version: Apple clang version 13.0.0 (clang-1300.0.29.3)

getwd: /Users/thomas
vmodules: /Users/thomas/.vmodules
vroot: /Users/thomas/Unix/v-latest
vexe: /Users/thomas/Unix/v-latest/v
vexe mtime: 2021-11-22 00:13:43
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 24ba660.e77a110

Git version: git version 2.34.0
Git vroot status: weekly.2021.46-64-ge77a1100
.git/config present: true
thirdparty/tcc status: thirdparty-macos-amd64 689c8a02

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg bug2.v`
```v
module main

interface TheInterface {
mut:
	an_interface()?
}

struct Implementation {
}

fn (mut i Implementation) an_interface ()? {
	return
}

fn maker () ?mut TheInterface {
	inner := &Implementation{}
	return *inner
}

fn do(mut inter TheInterface) {
}

fn main () {
	mut inner := maker()?
	do(mut *inner)
}
```

**What did you expect to see?**

the compiler reports the invalid pointer indirection used on `inner` when calling `do` and does not crash.

It seems there may be several other way to trigger the (perhaps same) issue

* changing the code to `fn maker () ?TheInterface {` make V detect the pointer issue and removing the indirection in `do` leads to another crash (look similar)

```v
module main

interface TheInterface {
mut:
	an_interface()?
}

struct Implementation {
}

fn (mut i Implementation) an_interface ()? {
	return
}

fn maker () ?TheInterface {
	inner := &Implementation{}
	return *inner
}

fn do(mut inter TheInterface) {
}

fn main () {
	mut inner := maker()?
	do(mut inner)
}
```

* changing the code to remove all optional also triggers a crash

```v
module main

interface TheInterface {
mut:
	an_interface()
}

struct Implementation {
}

fn (mut i Implementation) an_interface () {
	return
}

fn maker () TheInterface {
	inner := &Implementation{}
	return *inner
}

fn do(mut inter TheInterface) {
}

fn main () {
	mut inner := maker()
	do(mut inner)
}

```

**What did you see instead?**
```
==================
/tmp/v_501/bug2.488538216989206281.tmp.c:10694:11: error: passing 'main__TheInterface' (aka 'struct main__TheInterface') to parameter of incompatible type 'main__TheInterface *' (aka 'struct main__TheInterface *'); remove *
        main__do(*inner);
                 ^~~~~~
/tmp/v_501/bug2.488538216989206281.tmp.c:10682:61: note: passing argument to parameter 'inter' here
VV_LOCAL_SYMBOL main__Storage* main__do(main__TheInterface* inter) {
                                                            ^
1 error generated.
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/gen/c/fn.v
+++ b/vlib/v/gen/c/fn.v
@@ -1709,6 +1709,11 @@ fn (mut g Gen) ref_or_deref_arg(arg ast.CallArg, expected_type ast.Type, lang as
 				g.write('&/*sum*/')
 				g.expr(arg.expr)
 				return
+			} else if arg_sym.kind == .interface_ && exp_sym.kind == .interface_
+				&& (arg.expr is ast.Ident || arg.expr is ast.SelectorExpr) {
+				g.write('&/*iface*/')
+				g.expr(arg.expr)
+				return
 			}
 		}
 		if !g.is_json_fn {

--- a/vlib/v/tests/fn_mut_arg_of_interface_test.v
+++ b/vlib/v/tests/fn_mut_arg_of_interface_test.v
@@ -1,0 +1,27 @@
+interface TheInterface {
+mut:
+	an_interface() ?
+}
+
+struct Implementation {
+}
+
+fn (mut i Implementation) an_interface() ? {
+	return
+}
+
+fn maker() ?TheInterface {
+	inner := Implementation{}
+	return inner
+}
+
+fn do(mut inter TheInterface) string {
+	return 'ok'
+}
+
+fn test_fn_mut_arg_of_interface() ? {
+	mut inner := maker() ?
+	ret := do(mut inner)
+	println(ret)
+	assert ret == 'ok'
+}

